cookbooks/aws-parallelcluster-environment/recipes/install/cfn_bootstrap.rb

```diff
@@ -53,9 +53,9 @@
   creates "#{virtualenv_path}/bin/cfn-hup"
 end
 
-# Add cfn_bootstrap virtualenv to default path
-template "/etc/profile.d/pcluster.sh" do
-  source "cfn_bootstrap/pcluster.sh.erb"
+# Add cfn_bootstrap virtualenv to a profile to be used only for cookbook
+template "/etc/profile.d/cookbook_environment.sh" do
```

Comment:
If we define the cookbook environment as a script in profile.d, then it gets executed at every user login before the execution of pcluster.sh.  Is this the expected behaviour?

Correct me if I'm wrong but I think the goal here is to define the environment only for the cookbook, so this script should be placed outside profile.d and sourced at the beginning of the user data.

Also, since this is a pcluster specific file, I suggest to name it `pcluster_cookbook_environment.sh`.

Reply:
We are okay with the `cookbook_environment.sh` being in the profile.d as per the suggestion here https://github.com/aws/aws-parallelcluster/pull/6266#discussion_r1619169039

However, we want the PATH not being polluted as we were doing which affected every user. So we are now not changing the PATH but using a different variable

Reply:
Thanks for the clarification. 
I still think that this is not a good security practice. A script must be added to profile.d if and only if it must be loaded at every user login; otherwise it must be moved out of that directory.

Even if the script is currently safe to be loaded for every user it is still an unneeded vector of risks.